niddler-lib/src/main/java/com/icapps/niddler/core/NiddlerServiceLifeCycleWatcher.java

```diff
@@ -1,28 +1,72 @@
 package com.icapps.niddler.core;
 
 import android.app.Activity;
+import android.app.AlertDialog;
 import android.app.Application;
+import android.app.Dialog;
+import android.app.DialogFragment;
+import android.app.Fragment;
 import android.content.Context;
+import android.content.DialogInterface;
 import android.content.Intent;
 import android.content.ServiceConnection;
 import android.os.Bundle;
+import android.support.annotation.NonNull;
+
 import com.icapps.niddler.service.NiddlerService;
 
+import java.lang.ref.WeakReference;
+
+import static com.icapps.niddler.core.Niddler.INTENT_EXTRA_WAIT_FOR_DEBUGGER;
+
 /**
  * @author Nicola Verbeeck
- *         Date 22/11/16.
+ * Date 22/11/16.
  */
 class NiddlerServiceLifeCycleWatcher implements Application.ActivityLifecycleCallbacks {
 
+	private static final String WAIT_FOR_DEBUGGER_FRAGMENT_ID = "Niddler-Wait-For-Debugger";
+
+	@NonNull
 	private final ServiceConnection mServiceConnection;
+	@NonNull
+	private final Niddler mNiddler;
 
-	NiddlerServiceLifeCycleWatcher(final ServiceConnection connection) {
+	NiddlerServiceLifeCycleWatcher(@NonNull final ServiceConnection connection, @NonNull final Niddler niddler) {
 		mServiceConnection = connection;
+		mNiddler = niddler;
 	}
 
 	@Override
 	public void onActivityCreated(final Activity activity, final Bundle bundle) {
-		//Not handled
+		final int waitForDebugger = activity.getIntent().getIntExtra(INTENT_EXTRA_WAIT_FOR_DEBUGGER, 0);
+		if (waitForDebugger == 1) {
+
+			final WeakReference<Activity> weakActivity = new WeakReference<>(activity);
+
+			if (mNiddler.debugger().waitForConnection(new Runnable() {
+				@Override
+				public void run() {
+					final Activity startingActivity = weakActivity.get();
+					if (startingActivity != null) {
+						startingActivity.runOnUiThread(new Runnable() {
```

Comment:
Isn't it possible to use a lambda here?

Reply:
No java8 in libs, breaks java7 only projects